src/sparsezoo/v2/utils/backwards_compatibility.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A set of helper functions that serve
+as a temporary bridge between
+sparsezoo v1 and v2
+"""
+import copy
+from typing import Any, Dict, List, Set, Tuple
+
+
+__all__ = ["restructure_request_json", "fetch_from_request_json"]
+
+ALLOWED_FILE_TYPES = {
+    "originals",
+    "recipe",
+    "onnx",
+    "labels",
+    "card",
+    "inputs",
+    "training",
+    "deployment",
+    "benchmarking",
+    "outputs",
+}
+
+
+def restructure_request_json(
+    request_json: List[Dict[str, Any]], allowed_file_types: Set = ALLOWED_FILE_TYPES
+) -> Dict[str, Any]:
+    """
+    Takes the legacy API response and restructures it, so that the output is
+    compatible with the structure of ModelDirectory.
+
+    :params request_json: data structure describing the
+        files in the ModelDirectory (output from NeuralMagic API).
+    :params allowed_file_types: a set of `file_types`,
+        that will not be filtered out during restructuring
+    :return: restructured request_json
+    """
+    # create `training` folder
+    training_dicts_list = fetch_from_request_json(
+        request_json, "file_type", "framework"
+    )
+    for (idx, training_file_dict) in training_dicts_list:
+        training_file_dict["file_type"] = "training"
+        request_json[idx] = training_file_dict
+
+    # create `deployment` folder
+    onnx_model_dict_list = fetch_from_request_json(
+        request_json, "display_name", "model.onnx"
+    )
+    assert len(onnx_model_dict_list) == 1
+    _, onnx_model_file_dict = copy.copy(onnx_model_dict_list[0])
+    onnx_model_file_dict["file_type"] = "deployment"
+    request_json.append(onnx_model_file_dict)
+
+    # if NLP model, add `config.json` and `tokenizer.json` to `deployment`
+    training_file_names = [
+        x[1]["display_name"]
+        for x in fetch_from_request_json(request_json, "file_type", "training")
+    ]
+    nlp_folder = (
+        True
+        if (("config.json") in training_file_names)
+        and (("tokenizer.json") in training_file_names)
+        else False
+    )
```

Comment:
```suggestion
    nlp_folder = (
        "config.json" in training_file_names
        and "tokenizer.json" in training_file_names
    )
```

Reply:
Good one!

Reply:
I like that even more: `{"config.json", "tokenizer.json"}.issubset(set(training_file_names))`.
I've become  a set junkie lately.